### PR TITLE
vala: update to 0.42.4.

### PR DESCRIPTION
**Pull request by Cogitri**

### CODE REVIEW

**Comment by Cogitri** [comment]
Due to this moving usr/share/vala to vala-devel this may break packages which wrongly depend on vala instead of vala-devel.

File: srcpkgs/vala/template [hunk 1]
```diff
@@ -14,7 +14,8 @@ license="LGPL-2.1-or-later"
 homepage="https://wiki.gnome.org/Projects/Vala"
 changelog="https://gitlab.gnome.org/GNOME/vala/raw/0.42/NEWS"
 distfiles="${GNOME_SITE}/vala/${version%.*}/vala-${version}.tar.xz"
-checksum=5fc73dd1e683dc7391bb96d02b2f671aa8289411a48611a05265629e0048587d
+shlib_provides="libvalaccodegen.so"
```
Comment (maxice8) [inline srcpkgs/vala/template]: also needs to be in common/shlibs

Comment (Cogitri) [reply srcpkgs/vala/template]: Ah, right, forgot to push.

**Comment by Cogitri** [comment]
Alright, once CI is green this should be good to go.

**Comment by Cogitri** [comment]
Okie, this is good to go.

**Pull request merged**